src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,84 @@
+package duke.task;
+
+import duke.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A deadline task object that stores its description, the due date and on whether is it done or not. The task can be
+ * marked as done or unmarked as not done.
+ */
+public class Deadline extends Task {
+    /**
+     * The due date of the deadline task.
+     */
+    protected LocalDate by;
```

Comment:
Perhaps this instance attribute could be declared as private to better comply with Java information hiding.

Reply:
Okay thanks. Realised I missed it in my event class too.